broker/cloud_run/lsst/ps_to_storage/main.py

```diff
@@ -95,148 +84,62 @@ def run():
     Parameters
     ----------
-    raw : `bytes`
-        The 5-byte encoded message prefix.
+    alert : pittgoogle.Alert
+        The alert object.
     Returns
     -------
-    schema_version : `int`
-        A version number which indicates the Confluent Schema Registry ID
-        number of the Avro schema used to encode the message that follows this
-        header.
+    str: The formatted filename as "{schema_version}/{YYYY-MM-DD}/{diaObjectId}/{diaSourceId}.{format}".
     """
-    _, version = _ConfluentWireFormatHeader.unpack(raw)
+    time_obj = Time(alert.get("mjd"), format="mjd")
+    alert_date = time_obj.datetime.strftime(
+        "%Y-%m-%d"
+    )  # convert the MJD timestamp to "YYYY-MM-DD"
 
-    return version
+    return f"{alert.schema.version}/{alert_date}/{alert.objectid}/{alert.sourceid}.avro"
 
 
-def generate_alert_filename(aname: dict) -> str:
-    """
-    Generate the filename of an alert stored to a Cloud Storage bucket.
-
-    Args:
-        aname:
-            Components to create the filename. Required key/value pairs are those needed to create a parsed filename.
-            Extra keys are ignored.
-
-    Returns:
-        str: The formatted filename as "{schema_version}/{YYYY-MM-DD}/{objectId}/{sourceId}.{format}".
-    """
-
-    schema_version = aname["schema_version"]
-    alert_date = aname["alert_date"]
-    object_id = aname["objectId"]
-    source_id = aname["sourceId"]
-    file_format = aname["format"]
-
-    return f"{schema_version}/{alert_date}/{object_id}/{source_id}.{file_format}"
-
-
-def create_file_metadata(alert_dict: dict, event_id: str) -> dict:
+def _create_file_metadata(alert: pittgoogle.Alert, event_id: str) -> dict:
     """Return key/value pairs to be attached to the file as metadata."""
 
     metadata = {"file_origin_message_id": event_id}
-    metadata["diaObjectId"] = alert_dict["diaObject"]["diaObjectId"]
-    metadata["diaSourceId"] = alert_dict["diaSource"]["diaSourceId"]
-    metadata["ra"] = alert_dict["diaSource"]["ra"]
-    metadata["dec"] = alert_dict["diaSource"]["dec"]
+    metadata["diaObjectId"] = alert.objectid
+    metadata["diaSourceId"] = alert.sourceid
+    metadata["ra"] = alert.get("ra")
+    metadata["dec"] = alert.get("dec")
 
     return metadata
 
 
-def publish_alerts_stream(
-    topic_name: str, message: bytes, attributes: Optional[dict] = None
-) -> str:
-    """Publish original alert bytes to a Pub/Sub topic."""
-
-    # enforce bytes type for message
-    if not isinstance(message, bytes):
-        raise TypeError("`message` must be bytes.")
-
-    topic_path = publisher.topic_path(PROJECT_ID, topic_name)
-    future = publisher.publish(topic_path, data=message, **attributes)
-
-    return future.result()
-
-
-def _reformat_alert_data_to_valid_json(
-    alert_dict: dict, attributes: dict
-) -> pittgoogle.alert.Alert:
+def _reformat_alert_data_to_valid_json(alert: pittgoogle.Alert) -> dict:
```

Comment:
Do you think this is something `pittgoogle` should handle under the hood?

Reply:
Yes.

Reply:
Added to https://github.com/mwvgroup/pittgoogle-client/pull/81.